lib/iris/cube.py

```diff
@@ -991,6 +991,157 @@ def replace_coord(self, new_coord):
+                   "which is associated with {} dimensions")
+            msg = msg.format(aux_coord.name(), len(coord_dim))
+            raise ValueError(msg)
+
+        dim_coord = iris.coords.DimCoord.from_coord(aux_coord)
+
+        old_dim_coord = self.coords(dim_coords=True,
+                                    contains_dimension=coord_dim[0])
+
+        if len(old_dim_coord) == 1:
+            self.demote_dim_coord_to_aux_coord(old_dim_coord[0])
+
+        # order matters here: don't want to remove
+        # the aux_coord before have tried to make
+        # dim_coord in case that fails
+        self.remove_coord(aux_coord)
+
+        self.add_dim_coord(dim_coord, coord_dim)
+
+    def demote_dim_coord_to_aux_coord(self, name_or_coord):
+        """
+        Demotes a DimCoord on the cube to an AuxCoord, leaving that
+        dimension anonymous.
+
+        Args:
+
+        * name_or_coord:
+            Either
+
+            (a) An instance of :class:`iris.coords.DimCoord`
+
+            or
+
+            (b) the :attr:`standard_name`, :attr:`long_name`, or
+            :attr:`var_name` of an instance of an instance of
+            :class:`iris.coords.DimCoord`.
+
+        For example::
+
+            >>> print cube
+            air_temperature / (K)       (time: 12; latitude: 73; longitude: 96)
+                 Dimension coordinates:
+                      time                    x      -              -
+                      latitude                -      x              -
+                      longitude               -      -              x
+                 Auxiliary coordinates:
+                      year                    x      -              -
+            >>> cube.demote_dim_coord_to_aux_coord('time')
+            >>> print cube
+            air_temperature / (K)        (-- : 12; latitude: 73; longitude: 96)
+                 Dimension coordinates:
+                      latitude                -      x              -
+                      longitude               -      -              x
+                 Auxiliary coordinates:
+                      time                    x      -              -
+                      year                    x      -              -
+
+        """
+
+        if isinstance(name_or_coord, basestring):
+            dim_coord = self.coord(name_or_coord)
+        elif isinstance(name_or_coord, iris.coords.Coord):
+            dim_coord = name_or_coord
+        else:
+            # Don't know how to handle this type
+            msg = ("Don't know how to handle coordinate of type {}. "
+                   "Ensure all coordinates are of type basestring or "
+                   "iris.coords.Coord.")
+            msg = msg.format(type(name_or_coord))
+            raise TypeError(msg)
+
+        if not dim_coord in self.dim_coords:
+            # nothing to do
+            return
+
+        coord_dim = self.coord_dims(dim_coord)
+
+        aux_coord = iris.coords.AuxCoord.from_coord(dim_coord)
```

Comment:
(Confusingly!) You don't need to convert a DimCoord to an AuxCoord in order to add it to the aux_coords. Keeping it as a DimCoord would preserve the knowledge that it is monotonic.

In other words, I'd just do:
```python
self.remove_coord(dim_coord)
self.add_aux_coord(dim_coord, coord_dim)
```